lib/find_non_repeated_letter.rb

```diff
@@ -8,7 +8,29 @@
 #   nil (since there does not exist a non-repeated letter)
 
 
-def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
+def how_many_times_is_repeated(str)
+  #create hash to store key and value
+  h={}
+  #split up the string into individual characters and put it in an array
+  arr = str.split("")
+  #for each character in the array, if it does not have a previous occurence in the 'h' hash, add it to the hash with a value of '1'
+  arr.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+  #if it does have a previous occurence in the 'h' hash, add '1' to the previously occuring value of that key
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+    #return the completed hash
+    return h
```

Comment:
This `return h` is inside of the `arr.each` block. That means you're returning `h` one time for _each_ iteration of the loop.

You only need to do a `return` once–it's the final step of an algorithm.

So you've fixed the missing `end` problem, but the placement of that `end` is not correct.